Modules/App/Fixture/FixtureSpecification.cs

```diff
@@ -451,5 +463,44 @@ private FixtureFunction GetFunction(string functionName, FunctionIdentity functi
 		}
 
 		#endregion
+
+		#region IDeepCopy
+
+		/// <summary>
+		/// Refer to interface documentation.
+		/// </summary>		
+		public void Copy(object source)
+		{
+			// Get FixtureSpecification reference to the source object
+			FixtureSpecification src = (FixtureSpecification)source;
+			
+			// Copy the name of the fixture
+			Name = src.Name;
+
+			// Copy the manufacturer
+			Manufacturer = src.Manufacturer;
+
+			// Copy the name of the user that created the fixture profile
+			CreatedBy = src.CreatedBy;
+
+			// Copy the revision information 
+			Revision = src.Revision;
+
+			// Loop over the channel definitions
+			foreach (FixtureChannel channel in src.ChannelDefinitions)
+			{
+				// Clone the channel definition
+				ChannelDefinitions.Add(channel.CreateInstanceForClone());
+			}
+
+			// Loop over the function definitions
+			foreach (FixtureFunction function in src.FunctionDefinitions)
+			{
+				// Clone the function definition
+				FunctionDefinitions.Add(function.CreateInstanceForClone());
+			}		
+		}
+
+		#endregion
 	}
```

Comment:
Is there a reason why you would not just implement clone with deep copy logic internally?

Reply:
I may not understand your question.  
It seemed like the IObjectContentWriter patern's expectation was that WriteContentToObject method took a FixtureSpecification and expected the method to fill it up with data.  A Clone method would give me a new object and not assist with implementing this method.

Reply:
I think I see better what you were trying to do. I think the copy logic should just be in the XElementWriter instead of making purpose built interface and logic in the class for the copy. There should be an IFeatureSpecification interface that the writer can get as the type, then the writer should know how to populate the interface with the values that it deserializes. In essence the logic in copy would be in the writer and populate the interface object. That would closer follow the pattern of all the rest of them and keeps the model free of implementation that has no model value.

Reply:
Seems like to achieve what you are suggesting I have to move FixtureSpecification or IFixtureSpecification into the Vixen project.  Currently I don't have interfaces for the FixtureSpecification and any of its children.  Is it true that these reader/writer classes are trying to separate the persistence from the model?  In my case my model is a data contract so the persistence is baked in.  I need to sleep on this.

Reply:
I agree that your suggestion follows the existing pattern more closely.  I am wanting to avoid moving the FixtureSpecification classes to a new location in CM and a new namespace.  Seems like that change could ripple into a bunch of files.   Maybe if I did it all through the VS GUI it might not be too bad?  We have a C# architecture in my day job where most of the objects implement an IShallowCopy interface so it is not something foreign to me but I understand it how it might be foreign to you.

One thing I would defect myself on is that the existing CreateInstanceForClone should be calling (leveraging) the Copy method.  If I updated the clone method to call the Copy would that change your opinion at all since the method would then be providing model value?

Reply:
I wouldn't say the deep or shallow copy pattern is foreign. I have seen it numerous times. I was more looking to stay consistent with the existing patterns for consistency. I think I am ok with that being implemented internally and used for things like cloning. I think I am having issues with the naming of IDeepCopy. I am wondering if we can have something like IDataModel as the interface that could have some functions like CopyInto, and support cloning that might make this more palatable. There could be a more model like interface that could serve other concerns. Then the writer is working with something named more like a data model and is more self describing. 

Thoughts?

Reply:
I renamed the interface to IDataModel.  I renamed the method to CopyInto.  I also am leveraging the method inside the clone method.  Adjusted generic where clauses to require IDataModel.